main/zoe/guide/contract-installation-and-instance.md

```diff
@@ -0,0 +1,98 @@
+# Contract installation and instance
+
+Zoe has two methods to "create a contract": 
+- `zoe.install(code, moduleFormat)` which returns an `installationHandle`
+- `zoe.makeInstance(installationHandle, issuerKeywordRecord, terms)` which returns an `invite` to participate in the contract
+
+This two-step process to create may feel unnecessarily convoluted at first.
+
+The reason Zoe is designed this way is that it is assumed that the JavaScript source `code` of the contract may be large (several kilobytes, maybe several megabytes) and for the purpose of transparency, this source code will be stored in blockchains when Zoe runs on a blockchain.
```

Comment:
```suggestion
Zoe uses a two-step installation process because the JavaScript source `code` of the contract may be large (several kilobytes, maybe several megabytes). Because Zoe is intended to be run on a blockchain, storage is scarce and we want to avoid duplicate copies.
```

^ I think this is missing the second (potentially more important point), that we want people to be able to compare smart contracts easily. Comparing whether an installation handle matches another installation handle (object identity check) is as easy as it gets. The alternative would be asking someone to go line by line or do a git diff or something else to try to ensure that the contract someone is trying to get them to use is one they have used before and trust. 

Imagine that Alice uses the Atomic Swap contract regularly and finds it very useful. Then, Bob wants to do an Atomic Swap with Alice and sends her an invite. How does Alice know that the contract code that Bob uses is the code that she is used to? How does Alice prevent the attack in which most everything looks the same, but an essential condition is changed and she agrees to participate in something that she didn't intend? Having reusable `installationHandle`s solves this problem.